### PR TITLE
Python 3.7 -- including setuptools and pip

### PR DESCRIPTION
**Pull request by CameronNemo**
This allows a user to set up a python3.7 venv and use pip to install further dependencies without disrupting the system python3.6 package.

Self review is inline.

### CODE REVIEW

File: srcpkgs/python3.7-pip/template [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+license="MIT"
+homepage="https://pip.pypa.io/"
+distfiles="${PYPI_SITE}/p/pip/pip-${version}.tar.gz"
+checksum=c0a292bd977ef590379a3f05d7b7f65135487b67470f6281289a94e015650ea1
+alternatives="
+ pip:pip:/usr/bin/pip3.7
+ pip:pip3:/usr/bin/pip3.7"
+
+post_install() {
+	test ! -e "${DESTDIR}/usr/bin/pip3" || {
```
Comment (CameronNemo) [inline srcpkgs/python3.7-pip/template]: This should be handled by the build style, but that requires modifications that would affect python3.6 packages. Do we want to migrate to a new build style on the next python bump that uses the minor versions in the binary names?

File: srcpkgs/python3.7-setuptools/template [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ setuptools:easy_install:/usr/bin/easy_install3.7
+ setuptools:easy_install3:/usr/bin/easy_install3.7"
+provides="python3.7-distribute-${version}_1"
+replaces="python3.7-distribute>=0"
+
+export SETUPTOOLS_INSTALL_WINDOWS_SPECIFIC_FILES=0
+export SETUPTOOLS_DISABLE_VERSIONED_EASY_INSTALL_SCRIPT=1
+
+post_install() {
+	test ! -e "${DESTDIR}/usr/bin/easy_install" || {
```
Comment (CameronNemo) [inline srcpkgs/python3.7-setuptools/template]: ditto above

File: srcpkgs/python3.7-tkinter/template [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	fi
+	mv ${wrksrc}/${pkgname}-${version}-install/usr/bin/idle${version%.*} \
+		${DESTDIR}/usr/bin/idle${version%.*}
+	for lib in idlelib tkinter turtledemo turtle.py; do
+		mv ${wrksrc}/${pkgname}-${version}-install/usr/lib/python${version%.*}/${lib} \
+			${DESTDIR}/usr/lib/python${version%.*}/
+	done
+	mv ${wrksrc}/${pkgname}-${version}-install/usr/${_libdir}/python${version%.*}/lib-dynload/_tkinter.cpython-*m*.so \
+		${DESTDIR}/usr/lib/python${version%.*}/lib-dynload/
+	ln -sf idle${version%.*} ${DESTDIR}/usr/bin/idle3
```
Comment (CameronNemo) [inline srcpkgs/python3.7-tkinter/template]: probably should not ship `idle3` here and just use an alternative

File: srcpkgs/python3.7-tkinter/update [hunk 4]
```diff
@@ -0,0 +1 @@
+../python3/update
```
Comment (CameronNemo) [inline srcpkgs/python3.7-tkinter/update]: oops, this should be python3.7

File: srcpkgs/python3.7/template [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+makedepends="libffi-devel readline-devel gdbm-devel libressl-devel expat-devel
+ sqlite-devel bzip2-devel zlib-devel liblzma-devel"
+depends="ca-certificates"
+
+alternatives="
+ python:idle:/usr/bin/idle${version%.*}
+ python:2to3:/usr/bin/2to3-${version%.*}
+ python:pydoc:/usr/bin/pydoc${version%.*}
+ python:python:/usr/bin/python${version%.*}
+ python:python.1:/usr/share/man/man1/python${version%.*}.1"
```
Comment (CameronNemo) [inline srcpkgs/python3.7/template]: Need additional python3 alts here.

File: srcpkgs/python3.7/template [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+	if [ "$CROSS_BUILD" ]; then
+		# Remove references to cross toolchain.
+		sed -i "s/$XBPS_CROSS_TRIPLET-//g" \
+			${DESTDIR}/usr/lib/python${version%.*}/_sysconfigdata_m_*.py \
+			${DESTDIR}/usr/lib/python${version%.*}/config-${version%.*}m*/Makefile
+	fi
+}
+
+post_install() {
+	rm -f -- "${DESTDIR}/usr/lib/libpython3.so"
```
Comment (CameronNemo) [inline srcpkgs/python3.7/template]: this lib does not seem to be used by anyone or anything: `for pkg in $(xbps-query -RX python3); do xbps-query -RS "$pkg" | grep libpython3.so; done`

File: srcpkgs/python3.7-pip/template (hunk 1 above)
Comment (CameronNemo) [reply srcpkgs/python3.7-pip/template]: I just got rid of the entry points -- who really *needs* them?

**Comment by leahneukirchen** [comment]
We'll have Python 3.8 soon (tm), let's not bother with this.

**Pull request closed without merging**